File: FrontEnd-Mobile/app/src/main/java/com/it342/projectmanagementsystem/models/Tag.java
package com.it342.projectmanagementsystem.models;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.firebase.firestore.Exclude;

public class Tag implements Parcelable {
    private String name;
    private String color;

    // Default constructor required for Firestore
    public Tag() {
    }

    public Tag(String name, String color) {
        this.name = name;
        this.color = color;
    }

    protected Tag(Parcel in) {
        name = in.readString();
        color = in.readString();
    }

    public static final Creator<Tag> CREATOR = new Creator<Tag>() {
        @Override
        public Tag createFromParcel(Parcel in) {
            return new Tag(in);
        }

        @Override
        public Tag[] newArray(int size) {
            return new Tag[size];
        }
    };

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeString(color);
    }
} 

File: FrontEnd-Mobile/app/src/main/java/com/it342/projectmanagementsystem/models/Tag.kt
package com.it342.projectmanagementsystem.models

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

/**
 * Simple Tag class that will work with both Java and Kotlin
 */
@Parcelize
data class Tag(
    var name: String? = null,
    var color: String? = null
) : Parcelable {
    // No-arg constructor for Firebase
    constructor() : this(null, null)
    
    // Explicit getter and setter methods for Java compatibility with unique JVM names
    @JvmName("getNameForJava")
    fun getName(): String? {
        return name
    }
    
    @JvmName("setNameForJava")
    fun setName(name: String?) {
        this.name = name
    }
    
    @JvmName("getColorForJava")
    fun getColor(): String? {
        return color
    }
    
    @JvmName("setColorForJava")
    fun setColor(color: String?) {
        this.color = color
    }
} 